### ISSUE TITLE
Debug test lens

### ISSUE THREAD

**Issue by hdevalke** [rust-analyzer/rust-analyzer#3539]
I am the maintainer of the [Rust Test Lens](https://marketplace.visualstudio.com/items?itemName=hdevalke.rust-test-lens) extension in vscode.

[Issue 17](https://github.com/hdevalke/rust-test-lens/issues/17) was created because it was not possible to debug crates that contain a `main.rs` and a `lib.rs` in the same project. I cannot solve that problem without parsing the rust code.

As a proof of concept I tried to implement this in rust-analyzer see: https://github.com/hdevalke/rust-analyzer/commit/2e2ad8c1a5676efdd2cf1c0ef4acc17e0f7043c3.

Do you think it makes sense to integrate this in `rust-analyzer`, knowing that currently it depends on the [vscode-lldb](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb) extension?

**Comment by Veetaha** [comment]
Hi! Thank you for the awesome extension! It does help a lot to debug rust-analyzer tests itself! 

I personally don't see any impediments to merge that PoC, since `rust-analyzer` is not likely to replace `lldb` extension and is rather to be used in combination with it.
We had some discussion on this topic with @matklad at [this zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/185405-t-compiler.2Fwg-rls-2.2E0/topic/vscode_debug/near/186629693).

**Comment by matklad** [comment]
Yup, we should just do that, thanks for exploring this @hdevalke!

Although, impl-wise, I think we should try to avoid providing separate debug/run commands, and instead have a single command which can be both debugged and runed. Ie, instead of introducing `DebugConfiguration`, I would rather tweak `Runnable` two contain both the info required for run, and the info required for debug.

**Comment by matklad** [comment]
We also should upstream this to the LSP/VS Code. There should be a proper protocol request for telling which things can be run from a given context....

**Comment by matklad** [comment]
Opened https://github.com/microsoft/language-server-protocol/issues/944 upstream

**Comment by hdevalke** [comment]
Hi @matklad,

You are right sharing a common structure for debugging or running should work as well.
I did split up the `args` into `args` and `extra_args` because that's what the vscode-lldb debug configuration expects it and I did not had to search for `--` in the args vector.

The only difference between runnable and debug configuration is:
for non `bin` targets: add `--no-run`, for `bin` target replace `run` with `build`.

I made a new commit using only the modified `Runnable` struct: https://github.com/rust-analyzer/rust-analyzer/compare/0714a06...hdevalke:debug-lens.

To make this user friendly there is probably some need to add vscode-lldb as an extension dependency, or show a popup to propose to install vscode-lldb if `debug.startDebugging` returned `false`.
